clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -484,6 +548,25 @@ public void run() {
         return Collections.unmodifiableMap(childrenSensors);
     }
 
+    public MetricName metricInstance(MetricNameTemplate template, String... keyValue) {
```

Comment:
I had wanted to emphasize that it's an instance of a pre-defined metric, rather than one that is created on the fly. But, I suppose that I could rename it to metricName, and it would just be an overloaded function. My hope is that, after I have converted everything over, that MetricNameTemplates would be the preferred way of adding new metrics. Would it be too confusing to have metricName() be overloaded?

In https://github.com/apache/kafka/pull/1202, @junrao even asked if it would be possible to deprecate metricName(). I think I'd like to do that, but I'm not yet sure if that's possible. For example, in Fetcher.java, I wasn't able to switch the KIP-92 per-partition metrics over to MetricNameTemplates, because the names of the metrics are not entirely pre-defined.